pkgs/development/tools/geckodriver/default.nix

```
@@ -3,27 +3,28 @@
 , rustPlatform
 , stdenv
 , darwin
+, libiconv
 }:
 
 rustPlatform.buildRustPackage {
-  version = "0.26.0";
+  version = "0.29.1";
   pname = "geckodriver";
   sourceRoot = "source/testing/geckodriver";
 
   # Source revisions are noted alongside the binary releases:
   # https://github.com/mozilla/geckodriver/releases
   src = (fetchzip {
-    url = "https://hg.mozilla.org/mozilla-central/archive/e9783a644016aa9b317887076618425586730d73.zip/testing";
-    sha256 = "0m86hqyq1jrr49jkc8mnlmx4bdq281hyxhcrrzacyv20nlqwvd8v";
+    url = "https://hg.mozilla.org/mozilla-central/archive/970ef713fe58cbc8a29bfb2fb452a57e010bdb08.zip/testing";
+    sha256 = "0cpx0kx8asqkmz2nyanbmcvhnrsksgd6jp3wlcd0maid3qbyw7s2";
   }).overrideAttrs (_: {
     # normally guessed by the url's file extension, force it to unpack properly
     unpackCmd = "unzip $curSrc";
   });
 
   cargoPatches = [ ./cargo-lock.patch ];
-  cargoSha256 = "1dv8vcjy8r9z19grj4gms05bhaafyr42y3q69h3azwq6dmacfd3y";
+  cargoSha256 = "1fxm4jgg95igqadi97dzhrw7bhi5hfh7ww81c3z4yk0lznbzv6mx";
 
-  buildInputs = lib.optionals stdenv.isDarwin [ darwin.apple_sdk.frameworks.Security ];
+  buildInputs = [ libiconv ] ++ lib.optionals stdenv.isDarwin [ darwin.apple_sdk.frameworks.Security ];
```

```suggestion
  buildInputs = lib.optionals stdenv.isDarwin [ libiconv darwin.apple_sdk.frameworks.Security ];
```

libiconv is often a Darwin only dependency. This builds fine without it on Linux.

---

```suggestion
  buildInputs = lib.optionals stdenv.isDarwin [ libiconv Security ];
```
darwin shouldn't be an input as it cannot be easily overwriten.

---

@SuperSandro2000: I can't make `Security` an input here, at least not just by updating the parameters at the top. Any idea how I can make it injectable directly?

---

See for example: https://github.com/NixOS/nixpkgs/blob/0c896e754d968655826d4eeb319dde68cfb33b39/pkgs/top-level/all-packages.nix#L746-L748

---

That works. Pushed, thanks!